### PR TITLE
disable default admin, add reset and ensure admin command

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
What this PR does:
1. Added `CATTLE_NO_DEFAULT_ADMIN` settings so that rancher will not generate default admin and password.
2. Added reset-admin/ensure-admin sub-commands in rancherd so that it will generate/reset random password for rancher deployment. Server URL, Username and password will be printed out on CLI.

File: cmd/rancherd/go.mod [hunk 1]
```diff
@@ -61,7 +62,16 @@ replace (
 )
 
 require (
+	github.com/docker/docker v1.4.2-0.20191205034852-d163fbba3c82
+	github.com/natefinch/lumberjack v2.0.0+incompatible
+	github.com/pkg/errors v0.9.1
+	github.com/rancher/k3s v1.18.3-0.20200828235740-866dc94ceaac
 	github.com/rancher/rke2 v1.18.4-beta16.0.20200901184550-9c11691024ae
 	github.com/rancher/spur v0.0.0-20200617165101-8702c8e4ce7a
+	github.com/rancher/wrangler v0.6.1
```
Comment (ibuildthecloud) [inline cmd/rancherd/go.mod:71]: can you bump to master for wrangler

File: pkg/data/management/role_data.go [hunk 2]
```diff
@@ -383,6 +384,9 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
 // bootstrapAdmin checks if the bootstrapAdminConfig exists, if it does this indicates rancher has
 // already created the admin user and should not attempt it again. Otherwise attempt to create the admin.
 func BootstrapAdmin(management *wrangler.Context, createClusterRoleBinding bool) (string, error) {
+	if settings.NoDefaultAdmin.Get() != "" {
```
Comment (ibuildthecloud) [inline pkg/data/management/role_data.go:387]: Can you make this != "false"

Comment (ibuildthecloud) [inline pkg/data/management/role_data.go:387]:
```suggestion
	if settings.NoDefaultAdmin.Get() = "true" {
```

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/28131

**Pull request merged**